sgkit/utils.py

```diff
@@ -64,3 +64,55 @@ def encode_array(x: ArrayLike) -> Tuple[ArrayLike, List[Any]]:
     rank = np.empty_like(index)
     rank[index] = np.arange(len(index))
     return rank[inverse], names[index]
+
+
+def split_array_chunks(n: int, blocks: int) -> List[int]:
+    """Compute chunk sizes for an array split into blocks.
+    This is similar to `numpy.split_array` except that it
+    will compute the sizes of the resulting splits rather
+    than explicitly partitioning an array.
+    Parameters
+    ----------
+    n : int
+        Number of array elements.
+    blocks : int
+        Number of partitions to generate chunk sizes for.
+    Examples
+    --------
+    >>> split_array_chunks(7, 2)
+    [4, 3]
+    >>> split_array_chunks(7, 3)
+    [3, 2, 2]
+    >>> split_array_chunks(7, 1)
+    [7]
+    >>> split_array_chunks(7, 7)
+    [1, 1, 1, 1, 1, 1, 1]
+
+    Raises
+    ------
+    ValueError
```

Comment:
Could this be stuck into a single `ValuerError`?

Reply:
In the code, docs or both?  I think I've seen a couple conventions for multiple conditions like that but I'm not sure what will work with Sphinx -- maybe something like this? https://github.com/dask/dask/blob/master/dask/array/core.py#L3320

```
    ValueError
        * If list depths are mismatched - for instance, ``[[a, b], c]`` is
          illegal, and should be spelt ``[[a, b], [c]]``
        * If lists are empty - for instance, ``[[a, b], []]``
```

Reply:
Just in the doc.

Reply:
https://github.com/pystatgen/sgkit/pull/69/commits/bda112f96279b909983e3b9c7e47234d6ea9a973